packages/forms/src/Components/BaseFileUpload.php

```diff
@@ -99,6 +101,13 @@ protected function setUp(): void
             // An ugly mess as we need to support both Flysystem v1 and v3.
             $storageAdapter = method_exists($storage, 'getAdapter') ? $storage->getAdapter() : (method_exists($storageDriver = $storage->getDriver(), 'getAdapter') ? $storageDriver->getAdapter() : null);
             $supportsTemporaryUrls = method_exists($storageAdapter, 'temporaryUrl') || method_exists($storageAdapter, 'getTemporaryUrl');
+            
+            try {
+                $storage->getVisibility($file);
+            } catch (UnableToRetrieveMetadata $e) {
+                Log::error($e);
+                return '';
+            }
 
             if ($storage->getVisibility($file) === 'private' && $supportsTemporaryUrls) {
```

Comment:
See above comment. Does this work instead? I haven't tested it.

```suggestion
            if ($storage->exists($file) && $storage->getVisibility($file) === 'private' && $supportsTemporaryUrls) {
```